cmd/asdf/main.go

```diff
@@ -1,12 +1,31 @@
 // Main entrypoint for the CLI app
 package main
 
-import "github.com/asdf-vm/asdf/internal/cli"
+import (
+	"fmt"
+	"runtime/debug"
+	"slices"
 
-// Replaced with the real version during a typical build
-var version = "v-dev"
+	"github.com/asdf-vm/asdf/internal/cli"
+)
+
+// Do not touch this next line
+var version = "0.16.7" // x-release-please-version
 
 // Placeholder for the real code
 func main() {
-	cli.Execute(version)
+	fullVersion := buildFullVersion(version)
+	cli.Execute(fullVersion)
+}
+
+func buildFullVersion(version string) string {
+	info, ok := debug.ReadBuildInfo()
+	if !ok {
+		panic("can't get build info")
+	}
+	idx := slices.IndexFunc(info.Settings, func(s debug.BuildSetting) bool {
+		return s.Key == "vcs.revision"
+	})
+
+	return fmt.Sprintf("%s (revision %s)", version, info.Settings[idx].Value[0:7])
```

Comment:
The Git revision is important as it disambiguates dev builds of the master branch. For example, I merged this code and someone built latest master `asdf version` would print something like:

```
0.16.7 (revision 610d084)
```

But the `0.16.7` tag (if this logic were backported) would actually print:

```
0.16.7 (revision cbe8c6b)
```

Most users won't be building asdf on their own. But for those that do and then open up issues we'll be able to uniquely identify the commit they've build based on the revision.